src/test/java/org/embulk/input/mongodb/TestMongodbInputPlugin.java

```diff
@@ -698,7 +717,7 @@ private void assertValidRecords(Schema schema, MockPageOutput output, int limit,
         mapper.setDateFormat(getUTCDateFormat());
 
         int recordIndex = 0;
-        for (int i = skip; i < actualRecordSize; i++) {
+        for (int i = skip; i < maxRecordSize && i < skip + limit; i++) {
```

Comment:
This was actually incorrect before. The previous aggregation test had `skip == 4` and `actualRecordSize == 1` and wasn't actually entering the loop and testing anything.